src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -40149,6 +40151,416 @@ public void CopyTupleUnconstrainedElementNullability_02()
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "y.Item1").WithLocation(6, 9));
         }
 
+        [Fact]
+        public void CopyClassFields()
+        {
+            var source =
+@"#pragma warning disable 0649
+#pragma warning disable 8618
+class C
+{
+    internal object? F;
+    internal object G;
+}
+class Program
+{
+    static void F(C x)
+    {
+        if (x.F == null) return;
+        if (x.G != null) return;
+        C y = x;
+        x.F.ToString();
+        x.G.ToString(); // 1
+        y.F.ToString();
+        y.G.ToString(); // 2
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            // https://github.com/dotnet/roslyn/issues/32703: Not inferring nullability of non-nullable value compared to null.
+            // https://github.com/dotnet/roslyn/issues/31395: Nullability of class members should be copied on assignment.
```

Comment:
>// https://github.com/dotnet/roslyn/issues/31395: Nullability of class members should be copied on assignment. [](start = 12, length = 110)

If the current baseline doesn't meet what expected, consider adding a comment describing what is expected. This will make it easier to confirm in the future that the issue is addressed.